### PR TITLE
chore: updates to CI and the CI Examples

### PR DESCRIPTION
**Pull request by emilyrohrbough**
- Update CI pipelines & fix errors & hopefully the automatic release when merged to master
- Update CI examples to show pulling the latest node versions/images vs node which is now deprecated
- Bump the project's node version to 14

### CODE REVIEW

File: azure-ci.yml [hunk 1]
```diff
@@ -9,13 +9,13 @@ jobs:
   # Job names can contain alphanumeric characters and '_', cannot start with a number
   - job: Cypress_e2e_tests
     pool:
-      vmImage: 'ubuntu-16.04'
+      vmImage: 'ubuntu-latest'
```
Comment (emilyrohrbough) [inline azure-ci.yml:12]: Per https://github.com/actions/runner-images/issues/3287:
> Traditional 5-years support of Ubuntu 16.04 by Canonical [ends in April, 2021](https://ubuntu.com/16-04). To keep our environment updated and secured, we will remove Ubuntu 16.04 from GitHub on September 20, 2021, and from Azure DevOps on October 18, 2021.

This failed to run: https://cypress-io.visualstudio.com/cypress-example-kitchensink/_build/results?buildId=1642&view=logs&j=d2f18934-c168-59e0-ad20-3282681c6565

File: basic/.circleci/config.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@ version: 2
 jobs:
   test:
     docker:
-      - image: cypress/base:10
```
Comment (emilyrohrbough) [inline basic/.circleci/config.yml:5]: Cypress v12 dropped node 12 support. This is an example for users. Updated to pull latest.

File: basic/.gitlab-ci.yml [hunk 3]
```diff
@@ -13,7 +13,7 @@ cache:
     - cache/Cypress
 
 test:
-  image: cypress/base:10
```
Comment (emilyrohrbough) [inline basic/.gitlab-ci.yml:16]: Cypress v12 dropped node 12 support. This is an example for users. Updated to pull latest.

File: basic/.semaphore.yml [hunk 4]
```diff
@@ -20,7 +20,7 @@ blocks:
       jobs:
         - name: npm ci and cache
           commands:
-            - nvm install 12
```
Comment (emilyrohrbough) [inline basic/.semaphore.yml:23]: Cypress v12 dropped node 12 support. This is an example for users. Updated to use the latest LTS node version.

File: basic/.travis.yml [hunk 5]
```diff
@@ -1,8 +1,7 @@
 language: node_js
 
 node_js:
-  # Node 10.3+ includes npm@6 which has good "npm ci" command
-  - 10.8
+  - 18
```
Comment (emilyrohrbough) [inline basic/.travis.yml:4]: Cypress v12 dropped node 12 support. This is an example for users. Updated to use the latest LTS node version.

File: basic/azure-ci.yml [hunk 6]
```diff
@@ -9,7 +9,7 @@ jobs:
     steps:
       - task: NodeTool@0
         inputs:
-          versionSpec: '12.x'
+          versionSpec: '18.x'
```
Comment (emilyrohrbough) [inline basic/azure-ci.yml:12]: Cypress v12 dropped node 12 support. This is an example for users. Updated to use the latest LTS node version.

File: basic/codeship-pro/Dockerfile [hunk 7]
```diff
@@ -2,7 +2,7 @@
 # https://documentation.codeship.com/pro/languages-frameworks/nodejs/
 
 # use Cypress provided image with all dependencies included
-FROM cypress/base:10
+FROM cypress/base:latest
```
Comment (emilyrohrbough) [inline basic/codeship-pro/Dockerfile:5]: Cypress v12 dropped node 12 support. This is an example for users. Updated to pull latest.

File: basic/Jenkinsfile [hunk 8]
```diff
@@ -2,7 +2,7 @@ pipeline {
   agent {
     // this image provides everything needed to run Cypress
     docker {
-      image 'cypress/base:10'
+      image 'cypress/base:latest'
```
Comment (emilyrohrbough) [inline basic/Jenkinsfile:5]: Cypress v12 dropped node 12 support. This is an example for users. Updated to pull latest.

File: package.json [hunk 9]
```diff
@@ -77,7 +76,7 @@
     "yaml-lint": "1.2.4"
   },
   "engines": {
-    "node": ">=12"
+    "node": "^14.0.0 || ^16.0.0 || >=18.0.0"
```
Comment (emilyrohrbough) [inline package.json:79]: Match Cypress's node engines.

**Comment by emilyrohrbough** [comment]
The base build image needs updated for the [cypress-example-kitchensink](https://app.netlify.com/sites/cypress-example-kitchensink/deploys) project in netlify. The is currently is `Ubuntu 16.04 Xenial Xerus` which has been removed so the site is unable to deploy. It needs to be updated to `Ubuntu Focal 20.04`.

I don’t have access to the project settings to bump this myself. Have reached out to have this resolved by somone with access.

**Review by mjhenkes — APPROVED** [review]

**Pull request merged**